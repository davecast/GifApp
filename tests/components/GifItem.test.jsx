import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem/GifItem";

describe("Pruebas sobre <GifItem />", () => {
  const title = "Saitama";
  const url =
    "https://media1.giphy.com/media/n5j50VGDzkUqA/giphy.gif?cid=7a2ab4ce72ghst1qa9jnnjtvwe9rkp9jaw38nvc4nv2kq7j0&ep=v1_gifs_search&rid=giphy.gif&ct=g";

  test("should match with snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Should show the correctly image URL and ALT props", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);
    const { alt, src } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show the title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
