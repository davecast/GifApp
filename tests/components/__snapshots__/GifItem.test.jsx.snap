// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media1.giphy.com/media/n5j50VGDzkUqA/giphy.gif?cid=7a2ab4ce72ghst1qa9jnnjtvwe9rkp9jaw38nvc4nv2kq7j0&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
